theano/gpuarray/dnn.py

```diff
@@ -1868,9 +1872,18 @@ def local_gpua_pool_dnn_alternative(op, ctx_name, inputs, outputs):
     if not op.ignore_border:
         return
     img, ws, stride, pad = inputs
-    img = as_gpuarray_variable(img, ctx_name)
+    nd = op.ndim if op.ndim else (img.ndim - 2)
+    if nd not in (2, 3):
+        return
```

Comment:
No, I don't think so. Pool on the CPU supports arbitrary ndim, but the cuDNN version only supports 2D and 3D. This optimization has to check if dnn_pool can be used.